web/tls_config.go

```diff
@@ -203,7 +203,11 @@ func ServeMultiple(listeners []net.Listener, server *http.Server, flags *FlagCon
 // WebSystemdSocket in the FlagConfig is true. The FlagConfig is also passed on
 // to ServeMultiple.
 func ListenAndServe(server *http.Server, flags *FlagConfig, logger log.Logger) error {
-	if *flags.WebSystemdSocket {
+	if flags.WebSystemdSocket == nil && (flags.WebListenAddresses == nil || len(*flags.WebListenAddresses) == 0) {
+		return errors.New("no web listen address or systemd socket flag specified")
```

Comment:
Minor nit, I like to make these kinds of errors package vars so they can be used with `errors.Is()`.

Something like this:
```go
var ErrNoListeners = errors.New("...")
```

Reply:
I did a constant at the top if you're able to re-review :)